### PR TITLE
Allow issue templates to not render title

### PR DESCRIPTION
**Pull request by techknowlogick**
This adds a yaml attribute that will allow the option for when markdown is rendered that the title will be not included in the output

Based on work from @brechtvl

### CODE REVIEW

File: modules/issue/template/template_test.go [hunk 1]
```diff
@@ -640,6 +640,7 @@ body:
       description: Description of input
       placeholder: Placeholder of input
       value: Value of input
+      hide_label: true
```
Comment (delvh) [inline modules/issue/template/template_test.go:643]: Just as a question: Wouldn't omitting the `label` five lines above produce the same result?

**Comment by brechtvl** [comment]
Thanks! This works for us.

File: modules/issue/template/template.go [hunk 2]
```diff
@@ -311,6 +313,13 @@ func (f *valuedField) Label() string {
 	return ""
 }
 
+func (f *valuedField) HideLabel() bool {
+	if label, ok := f.Attributes["hide_label"].(bool); ok {
```
Comment (delvh) [inline modules/issue/template/template.go:317]: Hmm, I'm not quite sure if we should use snake case.
Also, shouldn't it be `hide_title`?

File: modules/issue/template/template_test.go (hunk 1 above)
Comment (techknowlogick) [reply modules/issue/template/template_test.go:643]: Yes, but in this situation you want to display the label on the form just not the output.

File: modules/issue/template/template.go (hunk 2 above)
Comment (techknowlogick) [reply modules/issue/template/template.go:317]: I was thinking the same with the name of the attr, but I went with hide_label to keep the name consistent throughout.

As for snakecase, the yaml marshal will make the map like this, and is consistent with other yaml parsing.

**Review by zeripath — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by jolheiser** [comment]
:trumpet: :robot:

**Comment by 6543** [comment]
adding it to the docs ->  #31574

**Pull request merged**